### ISSUE TITLE
Tabbing through suggestions of open and tabopen doesn't populate command-line

### ISSUE THREAD

**Issue by breggles** [cmcaine/tridactyl#420]
In Vimperator, when you tabbed through the suggestions when doing an `open` or `tabopen` it would populate the command-line with the URL of the suggestion so it could then be edited further, before opening it. Would be really good to get that functionality back.

Otherwise great addon, keep up the good work! 👍

**Comment by glacambre** [comment]
I think this was already mentioned in another issue but I can't find it.
You can currently add the completion option by selecting it with  tab and then pressing space. We're also planning on implementing the behavior you're describing once the command line has been rewritten.

**Comment by breggles** [comment]
Ah cool, thanks for letting me know the space bar workaround 😄

**Comment by cmcaine** [comment]
I think space to insert the completion is nicer, btw. Though we can support the other mode behind a setting.

**Comment by cmcaine** [comment]
Documentation is still not fantastic. If it was in the tutorial would you
have found it?

**Comment by bovine3dom** [comment]
> If it was in the tutorial, would you have found it?

Apparently not, as it is already in the tutorial on the command mode page 😃

**Comment by breggles** [comment]
😄 yeah only found out about the tutorial recently by accident by randomly reading one of the github issues. But it was a lot of words and i figured i already know most of the stuff from using Vimperator and Tridactyl for a while...

**Comment by breggles** [comment]
No ideas for the tutorial, in general, i'm afraid. For the issue at hand, i'd suggest auto-populating the command-line when you tab through, because it's just much more discoverable 😉

**Comment by stevenbell** [comment]
I'll second this. I can't really think of a downside to populating the URL; it would just sit empty otherwise.
One possible conflict is with flipping through history with the up/down arrows (which currently does populate the command line), but that should probably (a) close the tab-completions popup or (b) just use the arrow keys like tabs.

**Comment by ELLIOTTCABLE** [comment]
This is actually a really important issue to me; didn't know it was blocked by my work in #789! Damn, I'm gonna have to step it up. 😫

**Comment by ELLIOTTCABLE** [comment]
Oh, and can we stop adding a space to the end of the completions? Since the default keybind *is* <Space>, the user's finger will already be there if they need to type it again; and for URLs, it sucks quite a bit, because I keep having to do <Space> <Backspace> to start typing the next component of the URL I'm constructing!

**Comment by bovine3dom** [comment]
There's already history search - just type the first part of the string and keep pressing the up arrow.

I don't quite understand why you think this is blocked by #789.

**Comment by ELLIOTTCABLE** [comment]
“We're also planning on implementing the behavior you're describing once the command line has been rewritten”

Maybe I mis-interpreted!

**Comment by bovine3dom** [comment]
Ah, nope. That's rewriting the command line UI with iframes, completions etc - the parsing is tangential.

**Comment by ELLIOTTCABLE** [comment]
phew!

**Comment by glacambre** [comment]
>Can we at least add right-arrow to the fill-selected-completion keys?

You can do this yourself now with `bind --mode=ex <ArrowRight> ex.insert_completion`. You can also disable `<Space>` inserting completions with `unbind --mode=ex <Space>`.
